compiler/rustc_codegen_cranelift/src/driver/aot.rs

```diff
@@ -56,6 +56,21 @@ impl OngoingCodegen {
         backend_config: &BackendConfig,
     ) -> (CodegenResults, FxIndexMap<WorkProductId, WorkProduct>) {
         let mut work_products = FxIndexMap::default();
+
+        if !backend_config.disable_incr_cache {
```

Comment:
Reusing the crate metadata when disable_incr_cache is set is fine. This option exists to allow using incr comp while still recreating all object files. This allows for faster iteration of cg_clif changes by allowing the frontend results to be cached while still respecting changes to cg_clif.